Prog/LgUnit - Ipums Table 2B.sas

```diff
@@ -0,0 +1,543 @@
+/**************************************************************************
+ Program:  LgUnit - Ipums Table 2B.sas
+ Library:  IPUMS
+ Project:  NeighborhoodInfo DC
+ Author:   Yipeng and Rob
+ Created:  06/14/2018
+ Version:  SAS 9.4
+ Environment:  Local Windows session (desktop)
+ 
+ Description:  Use ipums data to calculate indicators for Large Units 
+			   table 2B. 
+
+ Modifications:
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( IPUMS )
+
+
+%let cvars = allhh largeunit isschoolage isadult iskid
+         issenior isdis raceW raceB raceH raceAPI raceO
+		 dcincome30 dcincome50 dcincome80 
+		 before1gen before2gen after1gen
+		 movedless1 moved2to10 moved10plus
+		 bedrooms0 bedrooms1 bedrooms2 bedrooms3 bedrooms4 bedrooms5plus
+		 units1 units2to4 units5to9 units10to19 units20plus
+		 BuiltBefore1940 Built1940to1959 Built1960to1979 Built1980to1999 Built2000After
+		 ;
+
+%let mvars = hher_age hh_inc kid_age isadult iskid;
+
+%macro recode_dvars (var1,var2);
+	if &var1. < 100 then &var2. = 0;
+	else if 100 <= &var1. < 200 then &var2. = 1;
+	else if 200 <= &var1. < 300 then &var2. = 2;
+	else if 300 <= &var1. < 400 then &var2. = 3;
+	else if 400 <= &var1. < 500 then &var2. = 4;
+	else if 500 <= &var1. < 600 then &var2. = 5;
+	else if 600 <= &var1. < 700 then &var2. = 6;
+	else if 700 <= &var1. < 800 then &var2. = 7;
+	else if 800 <= &var1. < 900 then &var2. = 8;
+	else if 900 <= &var1. < 1000 then &var2. = 9;
+	else if 1000 <= &var1. < 1100 then &var2. = 10;
+	else if 1100 <= &var1. < 1200 then &var2. = 11;
+	else if 1200 <= &var1. < 1300 then &var2. = 12;
+	else if 1300 <= &var1. < 1400 then &var2. = 13;
+	else if 1400 <= &var1. < 1500 then &var2. = 14;
+	else if 1500 <= &var1. < 1600 then &var2. = 15;
+%mend recode_dvars;
+
+
+%macro ipums_lgunit (tenure,yrs);
+
+%let ten = %upcase( &tenure. );
+%let yrs = %upcase( &yrs. );
+
+%if &yrs. = ACS %then %do;
+	%let indata = Acs_2012_16_dc;
+	%let vacdata = acs_2012_16_vacant_dc;
+%end;
+
+%if &yrs. = 2000 %then %do;
+	%let indata = Ipums_2000_dc;
+	%let vacdata = Ipums_2000_vacant_dc;
+%end;
+
+data vacdata;
+	set ipums.&vacdata. ;
+
+	%if &ten. = OWN %then %do;
```

Comment:
actually - it looks like you could use the rent or valueH to help determine rent vs. own?

Reply:
Yeah, used rent and valueh to code the ambiguous ones.

Reply:
@rpitingolo - i may be missing something but i have 3=rented or sold not occupied and 4= seasonal.  why is 3 only having to do with rental and 4 only ownership?

Reply:
@rpitingolo should we include other vacant too for all units?

Reply:
@lhendey ok, I updated these again. Now I'm using just codes 1 and 2 + the rent and valueh. I'm not sure about other vacant because I don't know exactly what's in there. The documentation isn't really clear on what it is either.